pkg/tfbridge/schema_test.go

```diff
@@ -583,10 +591,18 @@ func TestTerraformOutputsWithSecretsSupported(t *testing.T) {
 					},
 				},
 				f.NewSchemaMap(map[string]*schema.Schema{
-					// Type mapPropertyValue as a map so that keys aren't mangled in the usual way.
 					"float_property_value":     {Type: shim.TypeFloat},
 					"my_string_property_value": {Type: shim.TypeString},
-					"map_property_value":       {Type: shim.TypeMap},
+					"object_property_value": {
```

Comment:
```go
func TestTerraformInputSchemas(t *testing.T) {
	sch := map[string]*schemav2.Schema{
		// "schema_map": {Type: schemav2.TypeMap},  --> this is invalid
		"schema_map": {
			Type:     schemav2.TypeMap,
			Optional: true,
		},
		"unknown_array_value2": {
			Optional: true,
			Type:     schemav2.TypeList,
			MinItems: 1,
			Elem: (&schema.Resource{
				Schema: schemaMap(map[string]*schema.Schema{
					"required_property": {Type: shim.TypeString, Required: true},
					"conflicts_a":       {Type: shim.TypeString, ConflictsWith: []string{"conflicts_b"}},
					"conflicts_b":       {Type: shim.TypeString, ConflictsWith: []string{"conflicts_a"}},
				}),
			}).Shim(),
		},
		// INVALID: Type must be specified
		//
		// "object_property_value": {
		// 	Elem: &schemav2.Resource{
		// 		Schema: map[string]*schemav2.Schema{
		// 			"property_a": {Type: schemav2.TypeString, Required: true},
		// 			"property_b": {Type: schemav2.TypeBool, Optional: true},
		// 		},
		// 	},
		// },
	}
	err := schemav2.InternalMap(sch).InternalValidate(nil)
	require.NoError(t, err)
}

```

Reply:
Let's try to add test cases that are representable in TF under SDKv2 and therefore realistic. I'm checking all prior test cases against InternalValidate, not all of them pass, sometimes it's little things like Optional: true.

Reply:
This has now been addressed by https://github.com/pulumi/pulumi-terraform-bridge/pull/1853 and https://github.com/pulumi/pulumi-terraform-bridge/pull/1855, ensuring that all modified tests are `Validate`able. This PR has rebased on both.